docker/plate_solve_directory.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/python3
+import glob, os, sys, time
+import argparse
+import subprocess
+
+from panoptes.utils.images.fits import get_solve_field, fpack
+from panoptes.utils.error import Timeout
+
+from multiprocessing import Pool
+from functools import partial
+import tqdm
+import logging
+
+
+def solve_field_task(options, fpath):
+    """Small wrapper function to run get_solve_field() and ignore any timeouts.
+
+    Args:
+        options (dict): dictionary containing one keyword 'solve_opts' which maps
+                        to a list of solve-field arguments
+        fpath (str): The filepath of the file to be processed by solve-field
+    """
+    logger = logging.getLogger('main-log')
+    debug_logger = logging.getLogger('debug-log')
+    logger.info(f"Processing {fpath}")
+    if not os.path.isfile(fpath):
+        logger.warning(f"File does not exist, {fpath}")
+        return
+    if fpath.endswith('.fz'):
+        try:
+            fpath = fpack(fpath, unpack=True, overwrite=True)
+        except Exception as e:
+            debug_logger.warning(f"funpack failed on {fpath}, error: {e}")
+        if not os.path.isfile(fpath):
+            logger.warning(f"funpack failed on {fpath}")
+            return
+        else:
+            logger.info(f"Uncompressed file, output fpath is {fpath}")
+    logger.info(f"solving {fpath}")
+    try:
+        get_solve_field(fpath, **options)
+    except Timeout as e:
+        debug_logger.warning(f"get_solve_field() Timed out on {fpath} with error: \n{e}\n")
```

Comment:
Not sure we need to be adding newline characters for logger messages generally.

Reply:
The error messages that come out of solve_field errors can be several lines long, I find it hard to read the logs without the spaces

Reply:
In general with errors you want to format them `raw`, which should make it more readable:

```suggestion
        debug_logger.warning(f"get_solve_field() Timed out on {fpath} with error: {e!r}")
```